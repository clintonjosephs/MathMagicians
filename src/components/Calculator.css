.calculator {
  align-self: center;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr 1fr 1fr 1fr;
  width: 600px;
  height: 500px;
  font-size: 18px;
}

.screen {
  grid-column: span 4;
  display: flex;
  align-content: center;
  justify-content: flex-end;
  padding-top: 40px;
  padding-right: 5px;
  max-height: 60px;
  font-size: 25px;
  background: var(--primary-color);
  color: var(--white);
}

.zero {
  grid-column: span 2;
}

input[type="button"] {
  font-size: 20px;
  background: var(--neutral-color-1);
  border: 1px solid var(--neutral-color-2);
  position: relative;
  transition-duration: 0.4s;
}

input[type="button"]:active {
  background-color: var(--neutral-color-2);
  background-size: 100%;
  transition: background 0s;
}

.orange {
  background: var(--accent-color) !important;
}
