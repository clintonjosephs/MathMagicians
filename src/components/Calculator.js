import React, { useState } from 'react';
import { BiWinkSmile } from 'react-icons/bi';
import Screen from './controls/Screen';
import Button from './controls/Button';
import calculate from '../logic/calculate';
import CalculatorKeys from './controls/CalculatorKeys';
import './Calculator.css';

const Calculator = () => {
  const [data, setData] = useState({ total: '0' });

  const { total, operation, next } = data;

  const performOperation = (event) => {
    setData(calculate(data, event.target.value));
  };

  const buttons = CalculatorKeys();

  return (
    <div className="calculator-wrapper">
      <div
        style={{
          marginBottom: '20px',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        <BiWinkSmile style={{ fontSize: '40px' }} />
        <h2>Try out our calculator!</h2>
      </div>
      <div className="calculator">
        <Screen
          total={total}
          operation={operation}
          next={next}
        />
        {buttons.map(({ text, className }) => (
          <Button
            text={text}
            className={className}
            handlerClick={performOperation}
            key={text}
          />
        ))}
      </div>
    </div>
  );
};

export default Calculator;
