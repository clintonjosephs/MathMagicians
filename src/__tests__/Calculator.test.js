import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Calculator from '../components/Calculator';

describe('Use Jest snapshots to test the components.', () => {
  it('Renders calculator buttons', () => {
    const component = renderer.create(<Calculator />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Perform calculation operations', () => {
  test('Press to see response', () => {
    render(<Calculator />);
    const result = screen.getByTestId('screen');
    userEvent.click(screen.getByText('1'));
    expect(result.innerHTML).toBe('1');
  });

  test('Press two numbers to see response', () => {
    render(<Calculator />);
    userEvent.click(screen.getByText('1'));
    userEvent.click(screen.getByText('2'));
    const result = screen.getByTestId('screen');
    expect(result.innerHTML).toBe('12');
  });

  test('press equal sign to test calculation', () => {
    render(<Calculator />);
    userEvent.click(screen.getByText('AC'));
    userEvent.click(screen.getByText('1'));
    userEvent.click(screen.getByText('5'));
    userEvent.click(screen.getByText('+'));
    userEvent.click(screen.getByText('5'));
    userEvent.click(screen.getByText('='));
    const result = screen.getByTestId('screen');
    expect(result.innerHTML).toBe('20');
  });

  test('use calculator in modulus', () => {
    render(<Calculator />);
    userEvent.click(screen.getByText('AC'));
    userEvent.click(screen.getByText('1'));
    userEvent.click(screen.getByText('0'));
    userEvent.click(screen.getByText('%'));
    userEvent.click(screen.getByText('3'));
    userEvent.click(screen.getByText('='));
    const result = screen.getByTestId('screen');
    expect(result.innerHTML).toBe('1');
  });

  test('use calculator in multiplication', () => {
    render(<Calculator />);
    userEvent.click(screen.getByText('AC'));
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('0'));
    userEvent.click(screen.getByText('x'));
    userEvent.click(screen.getByText('3'));
    userEvent.click(screen.getByText('='));
    const result = screen.getByTestId('screen');
    expect(result.innerHTML).toBe('60');
  });
});
