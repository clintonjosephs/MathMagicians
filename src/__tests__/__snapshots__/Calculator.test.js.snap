// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use Jest snapshots to test the components. Renders calculator buttons 1`] = `
<div
  className="calculator-wrapper"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": "20px",
      }
    }
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "fontSize": "40px",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
      />
      <path
        d="M14.828 14.828a3.988 3.988 0 0 1-2.02 1.09 4.108 4.108 0 0 1-1.616 0 4.103 4.103 0 0 1-.749-.232 4.161 4.161 0 0 1-.679-.368 4.115 4.115 0 0 1-1.082-1.082l-1.658 1.117c.215.319.462.619.733.889a5.991 5.991 0 0 0 8.485.002c.272-.271.52-.571.734-.891l-1.658-1.117c-.143.211-.307.41-.49.592z"
      />
      <circle
        cx="8.5"
        cy="10.5"
        r="1.5"
      />
      <path
        d="M15.5 10c-2 0-2.5 2-2.5 2h5s-.501-2-2.5-2z"
      />
    </svg>
    <h2>
      Try out our calculator!
    </h2>
  </div>
  <div
    className="calculator"
  >
    <div
      className="screen"
      data-testid="screen"
    >
      0
    </div>
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="AC"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="+/-"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="%"
    />
    <input
      className="orange"
      onClick={[Function]}
      type="button"
      value="รท"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="7"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="8"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="9"
    />
    <input
      className="orange"
      onClick={[Function]}
      type="button"
      value="x"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="4"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="5"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="6"
    />
    <input
      className="orange"
      onClick={[Function]}
      type="button"
      value="-"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="1"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="2"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="3"
    />
    <input
      className="orange"
      onClick={[Function]}
      type="button"
      value="+"
    />
    <input
      className="zero"
      onClick={[Function]}
      type="button"
      value="0"
    />
    <input
      className=""
      onClick={[Function]}
      type="button"
      value="."
    />
    <input
      className="orange"
      onClick={[Function]}
      type="button"
      value="="
    />
  </div>
</div>
`;
